### PR TITLE
BAEL-5822 printstream vs printwriter

### PR DESCRIPTION
**Pull request by chrisjaimes**
1. Added main Java class for PrintStream and PrintWriter implementation
2. Added Unit Test Class
3. Added a (non-text) resource file for testing purposes to /resources directory

### CODE REVIEW

File: core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        File file = new File(dataProcessingTextFile);
+        assertTrue(file.exists());
+        assertEquals(FileUtils.readFileToString(file, "utf-8"), textFileContent);
+
+        Files.delete(Paths.get(dataProcessingTextFile));
+    }
+
+    @Test
+    public void giveFiles_writeNonTextDataWithStreams() throws IOException {
```
Comment (JavierPeris) [inline core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java:49]: The test method name must be: givenXXX_whenYYYY_thenZZZZ being the when and then mandatory and the given optional.

Comment (chrisjaimes) [reply core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java]: Fixed unit test names. Thanks for pointing that out.

File: core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    private static final String textFileContent = "Hello, world!";
+
+    @Test
+    public void whenUsingByteStream_thenWriteTextToFile() throws IOException {
+        DataStream dataStream = new DataStream();
+        dataStream.textDataProcessingByteStream(dataProcessingTextFile, textFileContent);
+
+        File file = new File(dataProcessingTextFile);
+        assertTrue(file.exists());
+        assertEquals(FileUtils.readFileToString(file, "utf-8"), textFileContent);
```
Comment (KevinGilmore) [inline core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java:31]: Swap the order of the arguments

File: core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test
+    public void whenUsingCharStream_thenWriteTextToFile() throws IOException {
+        DataStream dataStream = new DataStream();
+        dataStream.textDataProcessingCharStream(dataProcessingTextFile, textFileContent);
+
+        File file = new File(dataProcessingTextFile);
+        assertTrue(file.exists());
+        assertEquals(FileUtils.readFileToString(file, "utf-8"), textFileContent);
```
Comment (KevinGilmore) [inline core-java-modules/core-java-io-4/src/test/java/com/baeldung/iostreams/DataStreamUnitTest.java:43]: Swap the order of the arguments

**Review by KevinGilmore — COMMENTED** [review]
Swap the order of the arguments in assertEquals calls

**Pull request merged**